packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/methods/AnyMethod.kt

```diff
@@ -1,20 +1,41 @@
 package expo.modules.kotlin.methods
 
 import com.facebook.react.bridge.ReadableArray
-import expo.modules.core.Promise
+import expo.modules.kotlin.Promise
+import expo.modules.kotlin.exception.CodedException
+import expo.modules.kotlin.exception.InvalidArgsNumberException
+import expo.modules.kotlin.exception.UnexpectedException
 import expo.modules.kotlin.iterator
 import expo.modules.kotlin.types.TypeConverterHelper
+import kotlin.jvm.Throws
 import kotlin.reflect.KType
 
 abstract class AnyMethod(
   protected val name: String,
   private val desiredArgsTypes: Array<KType>
 ) {
-  abstract fun call(args: ReadableArray, promise: Promise)
+  fun call(args: ReadableArray, promise: Promise) {
+    if (desiredArgsTypes.size != args.size()) {
+      promise.reject(InvalidArgsNumberException(args.size(), desiredArgsTypes.size))
+      return
+    }
+    try {
+      val convertedArgs = convertArgs(args)
+      callUserImplementation(convertedArgs, promise)
+    } catch (codedError: CodedException) {
+      promise.reject(codedError)
+    } catch (e: Throwable) {
+      promise.reject(UnexpectedException(e))
+    }
+  }
+
+  @Throws(CodedException::class)
+  internal abstract fun callUserImplementation(args: Array<out Any?>, promise: Promise)
```

Comment:
`User` meaning is unclear in this context 🤔 
I would suggest getting rid of it or replacing the naming concept altogether.
```suggestion
  internal abstract fun callImplementation(args: Array<out Any?>, promise: Promise)
```
Alternatives: `doCall`, `invoke`.